### ISSUE TITLE
Is it possible to know when a home screen widget has been added?

### ISSUE THREAD

**Issue by roly151** [ABausG/home_widget#37]
Hi, 

At the moment I am registering a workmanager periodic task in my app's initstate to update my widget in the background. This works but it is registering & running the task even though there may be no widgets on the home screen for my app. I'm not sure if there is a way to know when the user adds a widget to the home screen so the workmanager task can be registered at that moment? Or is there another point I should be registering the workmanager task? Your example app uses a button, but that wouldn't work in this instance. So the outcome I'd like is to register the workmanager periodic task when a widget is added. 

Thankyou!

**Comment by Manuito83** [comment]
@ABausG I have encountered this problem as well. In my case, I'd need to decide whether I need to call an API every 15 minutes (which only makes sense if we have a widget active, as otherwise we would be wasting battery unnecessarily with the background worker).

On Android I believe we can override other several methods from AppWidgetProvider, such as `onDisabled`, `onEnabled`, etc. But this does not seem to be implemented. I'm not sure what's the case on iOS.

Is there a workaround for this in either platform? Otherwise, are there plans to implement this feature?

Thanks!

**Comment by kevinsamyn** [comment]
Hi @Manuito83 can you give me a link or an example of your solution ? Thanks.

**Comment by sabifa** [comment]
> @ABausG I have encountered this problem as well. In my case, I'd need to decide whether I need to call an API every 15 minutes (which only makes sense if we have a widget active, as otherwise we would be wasting battery unnecessarily with the background worker).
> 
> On Android I believe we can override other several methods from AppWidgetProvider, such as `onDisabled`, `onEnabled`, etc. But this does not seem to be implemented. I'm not sure what's the case on iOS.
> 
> Is there a workaround for this in either platform? Otherwise, are there plans to implement this feature?
> 
> Thanks!

We are facing the exact same problem. It would be nice to hear thoughts on this from the author of the lib @ABausG. 
This issue was tagged as "enhancement" long time ago, are there any plans to support this or is there no demand for it?

If this library does not focus on these challenges it would be nice to provide additional documentation on how to set this up correctly using native code.

@Manuito83 Did you end up with a solution that you would like to share?

**Comment by ABausG** [comment]
Hey,

thanks there has been a recent PR https://github.com/ABausG/home_widget/pull/120 which I had not yet have the time to review. Which seems to add the functionality at least to the point of knowing if this is working.
Would you have time to try out the changes made in the PR and give feedback in the PR about the API and general functionality of the new feature? If that is working and makes sense from a usability standpoint I am happy to incorporate that in the next release

**Comment by Ownii** [comment]
The PR adds an api to count the number of widgets. This does not notify that an widget has been added, so we can do things like start a periodic update task. We would be able to check if there is at least one widget at app startup but until then there would be no updates.

**Comment by richirisu** [comment]
I think it's technically possible to achieve this without polling but event triggered. Neither the AppWidgetManager on Android nor the WidgetCenter on iOS has an event to subsribe to. however, whenever a widget is added, it's update method should be called right away, from where it might be possible to invoke a callback to the Flutter side.

What is most likely not possible is to have an event for the case that a widget is being removed. But I haven't really confirmed this yet.

**Comment by lora-explora** [comment]
Hi, I faced the same issue. My workaround right now is to save a timestamp in my widgets (sharedprefs/userdefaults), when the code is executed. Then I can read the timestamp in my app and if the value is set and. not too old I schedule the background jobs.. I also posted a question on Stackoverflow, but no other methods so far.. https://stackoverflow.com/questions/77812000/how-can-i-find-out-if-a-user-installed-a-home-screen-widget-of-my-flutter-app/77848467#77848467
